Bottom: 6765a0c30671e9913abf7680b7ecc2c0c4415ffc
Top:    adbc5e044ca220d9d21d72e8b804f954a84df8bc
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-10-24 21:52:19 -0700

Refresh of rs-iomap

---

diff --git a/src/rsocket.c b/src/rsocket.c
index eb8b3f0..f0c4c35 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -2319,7 +2319,7 @@ off_t riomap(int socket, void *buf, size_t len, int prot, int flags, off_t offse
 	int access = IBV_ACCESS_LOCAL_WRITE;
 
 	rs = idm_at(&idm, socket);
-	if (rs->cm_id->pd || (prot & ~(PROT_WRITE | PROT_NONE)))
+	if (!rs->cm_id->pd || (prot & ~(PROT_WRITE | PROT_NONE)))
 		return ERR(EINVAL);
 
 	fastlock_acquire(&rs->iomap_lock);
